hips/hip-0015.md

```diff
@@ -0,0 +1,126 @@
+---
+hip: "0015"
+title: "Annotation for Images used by Helm Charts"
+authors:
+  [
+    "Tom Runyon <tom@defenseunicorns.com>",
+    "Scott Rigby <scott@weave.works>",
+    "Andrew Block <ablock@redhat.com>",
+  ]
+created: "2021-10-07"
+type: "feature"
+status: "draft"
+---
+
+## Abstract
+
+The images used by a Helm chart is not easily discoverable in the chart definition. When deploying Operators and CRDs, the images needed to instantate all the pods for a Helm release may not be present by templating the default values for a Helm chart, whic provides blockers when automatically discovering dependencies for Bill of Materials.
+
+## Motivation
+
+Without a standard methodology for defining the images used by a chart, third party applications that require metadata for Helm Charts are using custom keys unique to their application, e.g. [Artifact Hub](https://artifacthub.io/docs/topics/annotations/helm/#supported-annotations) or [Openshift](https://docs.openshift.com/container-platform/4.8/operators/operator_sdk/osdk-generating-csvs.html#olm-enabling-operator-for-restricted-network_osdk-generating-csvs). By Standardizing metadata, charts can be built to be consumable by a varity of tools for metadata, bill of materials, airgap deployments, etc. Current solutions of dynamically discovering the images needed by templating the chart do not account for all possible templating paths for the chart, nor capture images that are deployed as part of an Operator paradigm and not defined explicitly as pods in the helm chart.
+
+## Rationale
+
+The `helm.sh` prefix identifies the standard belonging to the Helm community, and the `/images` path clearly identifies the type of objects being identified. The annotation will define a string that can be parsed into a list that the chart maintainer will manage as they knowingly adjust the images needed for the chart's deployment. Each element of this image list will contain at least a `name` for the image as well as a location for the image in the `image` field. and an optional `dependency` field described in the specification. The elements _may_ have additional properties to meet additional use cases, such as the `whitelisted` field used by ArtifactHub, or providing a public key for signed images. The additional fields are not part of this standard, but how they get attached to each image _is_.
```

Comment:
Noticed in [this HIP](https://github.com/helm/community/pull/176) a request to override dependencies.  Maybe a helm option to provide a Chart.yaml overlay is the best solution to cover both scenarios and `appversion`.

Reply:
I would suggest focusing on images first instead of the other elements. The references to bom and signature point in a direction where security is in the loop. The references follow what's happening with sigstore. This should happen as part of a broader conversation (which I would like to have).